HeterogeneousCore/SonicTriton/src/TritonClient.cc

```diff
@@ -38,14 +38,16 @@ namespace {
 
 TritonClient::TritonClient(const edm::ParameterSet& params, const std::string& debugName)
     : SonicClient(params, debugName, "TritonClient"),
+      batchMode_(TritonBatchMode::Rectangular),
+      manualBatchMode_(false),
       verbose_(params.getUntrackedParameter<bool>("verbose")),
       useSharedMemory_(params.getUntrackedParameter<bool>("useSharedMemory")),
-      compressionAlgo_(getCompressionAlgo(params.getUntrackedParameter<std::string>("compression"))),
-      options_(params.getParameter<std::string>("modelName")) {
+      compressionAlgo_(getCompressionAlgo(params.getUntrackedParameter<std::string>("compression"))) {
+  options_.emplace_back(params.getParameter<std::string>("modelName"));

```

Comment:
Would
```suggestion
      options_{params.getParameter<std::string>("modelName"},
      compressionAlgo_(getCompressionAlgo(params.getUntrackedParameter<std::string>("compression"))) {
```
work?

Reply:
Unfortunately the constructor is declared as explicit: https://github.com/triton-inference-server/client/blob/r22.08/src/c%2B%2B/library/common.h#L160
So trying this gives a compiler error:
```
error: converting to 'triton::client::InferOptions' from initializer list would use explicit constructor 'triton::client::InferOptions::InferOptions(const string&)'
```

The only way I can get it to work is:
```
options_{tc::InferOptions(params.getParameter<std::string>("modelName"))}
```

Is that still preferable?

Reply:
Maybe the `emplace_back()` is more clear in this case.